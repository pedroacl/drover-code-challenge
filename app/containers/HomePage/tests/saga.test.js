import {
  loadCars as loadCarsSaga,
  loadMoreCars as loadMoreCarsSaga,
} from '../saga';

import { put, takeLatest } from 'redux-saga/effects';

import {
  loadCars as loadCarsAction,
  loadingCars as loadingCarsAction,
  loadMoreCars as loadMoreCarsAction,
  carsLoaded as carsLoadedAction,
} from '../actions';

describe('HomePage saga', () => {
  describe('loadCars Saga', () => {
    let loadCarsGenerator;
    let putLoadingCars;
    let callLoadCars;
    let putCarsLoaded;

    const carsRes = [
      {
        title: 'car1',
      },
    ];

    beforeAll(() => {
      const params = {
        location: 'London',
      };

      loadCarsGenerator = loadCarsSaga(loadCarsAction(params));

      putLoadingCars = loadCarsGenerator.next().value;
      expect(putLoadingCars).toMatchSnapshot();

      callLoadCars = loadCarsGenerator.next().value;
      expect(callLoadCars).toMatchSnapshot();

      putCarsLoaded = loadCarsGenerator.next(carsRes).value;
      expect(putCarsLoaded).toMatchSnapshot();
    });

    it('dispatches a loading cars action', () => {
      const location = 'London';
      expect(putLoadingCars).toEqual(put(loadingCarsAction(location)));
    });

    it('dispatches a call action', () => {
      // TODO add mock
    });

    it('dispatches a cars loaded action', () => {
      expect(putCarsLoaded).toEqual(put(carsLoadedAction(carsRes)));
    });
  });

  describe('loadMoreCars Saga', () => {
    let loadMoreCarsGenerator;

    beforeEach(() => {
      const params = {
        location: 'London',
      };

      loadMoreCarsGenerator = loadMoreCarsSaga(loadMoreCarsAction(params));

      const callDescriptor = loadMoreCarsGenerator.next().value;
      expect(callDescriptor).toMatchSnapshot();
    });

    it('loads more cars', () => {

    });
  });
});
