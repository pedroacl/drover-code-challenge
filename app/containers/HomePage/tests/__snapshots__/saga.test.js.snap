// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": "London",
      "type": "app/HomePage/actions/LOADING_CARS",
    },
    "channel": null,
  },
}
`;

exports[` 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[` 3`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Array [
        Object {
          "title": "car1",
        },
      ],
      "type": "app/HomePage/action/CARS_LOADED",
    },
    "channel": null,
  },
}
`;

exports[`HomePage saga loadMoreCars Saga loads more cars 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": "London",
      "type": "app/HomePage/actions/LOADING_MORE_CARS",
    },
    "channel": null,
  },
}
`;
